olp-cpp-sdk-dataservice-read/src/QueryMetadataJob.h

```diff
@@ -66,14 +67,12 @@ class QueryMetadataJob {
       QueryItemsFunc<ItemType, QueryType> query,
       FilterItemsFunc<ItemType> filter,
       std::shared_ptr<DownloadItemsJob<ItemType, PrefetchResult>> download_job,
-      std::shared_ptr<thread::TaskScheduler> task_scheduler,
-      std::shared_ptr<client::PendingRequests> pending_requests,
-      client::CancellationContext execution_context, uint32_t priority)
+      TaskSink& task_sink, client::CancellationContext execution_context,
+      thread::Priority priority)
```

Comment:
Why did you change it to `thread::Priority`? What happens if the user sets some other prio than the ones defined in the `thread::Priority` enum?

Reply:
reverted